lib/manager.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+let Manifest = require("./manifest");
+let { createFile, generateFingerprint, uriJoin } = require("faucet-pipeline-util");
+let path = require("path");
+
+module.exports = class AssetManager {
+	constructor(referenceDir, { manifestConfig, fingerprint, exitOnError } = {}) {
+		this.referenceDir = referenceDir;
+		this.manifestConfig = manifestConfig;
+		this.fingerprint = fingerprint;
+		this.exitOnError = exitOnError;
+		this.manifest = new Manifest();
+
+		// bind methods for convenience
+		this.writeFile = this.writeFile.bind(this);
+		this.resolvePath = this.resolvePath.bind(this);
+	}
+
+	writeFile(filepath, data, { error } = {}) {
+		let originalPath = filepath;
+		if(this.fingerprint) {
+			filepath = generateFingerprint(filepath, data);
+		}
+
+		return createFile(filepath, data).
+			then(_ => this._updateManifest(originalPath, filepath)).
+			then(_ => {
+				this._report(originalPath, error);
+				if(error && this.exitOnError) {
+					abort();
+				}
+			}).
+			catch(abort);
+	}
+
+	resolvePath(filepath, { enforceRelative } = {}) {
+		if(filepath.substr(0, 2) === "./") {
+			return path.resolve(this.referenceDir, filepath);
+		} else if(enforceRelative) {
+			throw new Error(`path must be relative: \`${repr(filepath)}\``);
+		} else { // attempt via Node resolution algorithm
+			try {
+				return resolveModulePath(filepath, this.referenceDir);
+			} catch(err) {
+				throw new Error(`could not resolve \`${repr(filepath)}\``);
+			}
+		}
+	}
+
+	_updateManifest(originalPath, actualPath) {
+		originalPath = path.relative(this.referenceDir, originalPath);
+		actualPath = path.relative(this.referenceDir, actualPath);
+
+		let config = this.manifestConfig;
+		let baseURI = config && config.baseURI;
```

Comment:
I think I would cut the responsibilities slightly differently:

The `Manifest` really takes care of the `Manifest`. The `AssetManager` doesn't need to know about a baseURI, that's the job of the Manifest. The `AssetManager` simply fingerprints & writes files and tells the Manifest it did so. This would also mean, you will just pass along the configuration for the manifest to the manifest class. 
  
TBQH I would move this entire `_updateManifest` method to the manifest, it has exactly the signature that I would like the `add` method to have.